content/documentation/pages/6-recipes/8-functional-apps/1-scst-function-bindings.md

```diff
@@ -0,0 +1,139 @@
+		public TimeSupplierConfiguration() {
+		}
+
+		@Bean
+		public Supplier<String> timeSupplier() {
+			SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS");
+			return () -> {
+				return sdf.format(new Date());
+			};
+		}
+	}
+}
+
+
+```
+
+Spring Cloud Stream provides `spring.cloud.stream.poller.DefaultPollerProperties` which can be configured to trigger the `Supplier` function `timeSupplier()` above.
+For instance, you can use `--spring.cloud.stream.poller.fixed-delay=5000` property to trigger this `Supplier` function every 5s.
+
+Similarly, You can write a `Sink` application using `java.util.function.Consumer` as follows:
+
+```
+package com.example.logsink;
+
+import java.util.function.Consumer;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.integration.handler.LoggingHandler;
+import org.springframework.messaging.Message;
+
+@Configuration
+public class LogConsumerConfiguration {
+	public LogConsumerConfiguration() {
+	}
+
+	@Bean
+	IntegrationFlow logConsumerFlow() {
+		return IntegrationFlows.from(MessageConsumer.class, (gateway) -> gateway.beanName("logConsumer"))
+				.handle((payload, headers) -> {
+					if (payload instanceof byte[]) {
+						return new String((byte[]) payload);
+					}
+					return payload;
+				})
+				.log(LoggingHandler.Level.INFO, "log-consumer", "payload")
+				.get();
+	}
+
+	private interface MessageConsumer extends Consumer<Message<?>> {}
+}
+
+```
+
+and,
+
+```
+package com.example.logsink;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class LogSinkApplication {
+
+	public static void main(String[] args) {
+		SpringApplication.run(LogSinkApplication.class, args);
+	}
+}
+
+```
+
+After building both the `time-source` and `log-sink` applications, you can register these applications into Spring Cloud Data Flow.
+
+Let's say you create a stream using these applications as follows:
+
+```
+ticktock=time-source | log-sink
+```
+
+You need to make sure that the functional bindings of these applications are mapped to the appropriate `output` and `input` names that Spring Cloud Data Flow understands.
+
+What this means is that, you need to configure the following properties when `deploying` the stream:
+
+```
+app.time-source.spring.cloud.stream.function.bindings.timeSupplier-out-0=output
+app.log-sink.spring.cloud.stream.function.bindings.logConsumer-in-0=input
+```
+
+The timeSupplier function's output and `logConsumer` function's input need to be mapped to the outbound/inbound names that Spring Cloud Data Flow understands: `output` and `input`.
+
+Along with this, you also need to provide a way trigger the `Supplier` function. In this case,
```

Comment:
Maybe move this to the DSL itself via `--fixed-delay=5`? One less step to discuss in the demo walkthrough.

Reply:
Since I didn't add the properties white-listing, I stuck to this approach. Otherwise, we would need to add the white-listing into the samples. What do you think?

Reply:
OK, let's go without it.